Fauna-RPG/Entity/Components/HealthComponent.gd

```diff
@@ -0,0 +1,72 @@
+extends EntityComponent
+class_name HealthComponent
+## Holds player health values and allows for dealing damage.
+
+
+signal on_health_changed(new_health : int)
+
+@export var max_health : float = 100
+var current_health : float = 100
+
+
+func _ready() -> void:
+	_set_health(max_health)
+
+
+# Allow attacker to be null, but not the victim.
+# If attacker is null, we assume that the damage came from an unknown source,
+# so buffs won't calculate.
+# Use a negative damage value if you want to heal.
+func deal_damage(damage_data : DealDamageData) -> void:
+	var damage = damage_data.damage
+
+	if damage == 0.0:
+		return
+	
+	assert(owner != null, "No owner was set. Please call init on the Entity.")
+	
+	var victim : Entity = entity_owner
+	var attacker : Entity = damage_data.attacker
+	
+	# if this was a self attack, ignore the attacker
+	if attacker == victim: 
```

Comment:
Maybe some effects will make it so that you want to attack yourself ? Like deal 3 damage to all enemies every time you take damage ?
Maybe not relevant right now, but to keep in mind for later

Reply:
Hmm maybe. I think it's a bit weird design wise that a player can attack themselves, but something to consider